osu.Game/Screens/Utility/LatencyArea.cs

```diff
@@ -0,0 +1,239 @@
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input;
+using osu.Framework.Input.Events;
+using osu.Game.Overlays;
+using osuTK;
+using osuTK.Input;
+
+namespace osu.Game.Screens.Utility
+{
+    public class LatencyArea : CompositeDrawable
+    {
+        [Resolved]
+        private OverlayColourProvider overlayColourProvider { get; set; } = null!;
+
+        public Action? ReportUserBest { get; set; }
+
+        private Drawable? background;
+
+        private readonly Key key;
+
+        public readonly int TargetFrameRate;
+
+        public readonly BindableBool IsActiveArea = new BindableBool();
+
+        public LatencyArea(Key key, int targetFrameRate)
+        {
+            this.key = key;
+            TargetFrameRate = targetFrameRate;
+
+            RelativeSizeAxes = Axes.Both;
+            Masking = true;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            InternalChildren = new[]
+            {
+                background = new Box
+                {
+                    Colour = overlayColourProvider.Background6,
+                    RelativeSizeAxes = Axes.Both,
+                },
+                new ButtonWithKeyBind(key)
+                {
+                    Text = "Feels better",
+                    Y = 20,
+                    Width = 0.8f,
+                    Anchor = Anchor.TopCentre,
+                    Origin = Anchor.TopCentre,
+                    Action = () => ReportUserBest?.Invoke(),
+                },
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        new LatencyMovableBox(IsActiveArea)
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                        },
+                        new LatencyCursorContainer(IsActiveArea)
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                        },
+                    }
+                },
+            };
+
+            IsActiveArea.BindValueChanged(active =>
+            {
+                background.FadeColour(active.NewValue ? overlayColourProvider.Background4 : overlayColourProvider.Background6, 200, Easing.OutQuint);
+            }, true);
+        }
+
+        protected override bool OnMouseMove(MouseMoveEvent e)
+        {
+            IsActiveArea.Value = true;
+            return base.OnMouseMove(e);
+        }
+
+        private double lastFrameTime;
+
+        public override bool UpdateSubTree()
+        {
+            double elapsed = Clock.CurrentTime - lastFrameTime;
```

Comment:
Is there a reason to recompute locally rather than use `Clock.ElapsedFrameTime`? I get why `LatencyMovableBox` can't use that (it can't because the latency is simulated by the drawable not receiving updates from its parent, so it can't just read the value from the clock because the clock is running normally), but this should be able to...?

Reply:
Very correct.

Reply:
Actually not really. I would need to create a sum variable and add up elapsed time if I was to remove this.